src/integration_tests/rpc_multiple_remote_addrs/main.cc

```diff
@@ -124,26 +140,27 @@ main(int args, char **argv, char **env) {
         return rpc.invoke_on_all(&smf::rpc_server::start);
       })
       .then([random_port] {
-        return seastar::do_with(clients(random_port), [] (auto& clients) {
-          return seastar::do_for_each(clients, [] (auto& client) {
+        return seastar::do_with(clients(random_port), [](auto &clients) {
+          return seastar::do_for_each(clients, [](auto &client) {
             DLOG_DEBUG("Connecting client to {}", client.client->server_addr);
             return client.client->connect()
-            .then([&client] {
-              DLOG_DEBUG("Connected to {}", client.ip_str);
-              smf::rpc_typed_envelope<smf_gen::demo::Request> req;
-              req.data->name = seastar::sstring(client.ip_str, INET_ADDRSTRLEN);
-              return client.client->Get(req.serialize_data());
-            })
-            .then([&client] (auto reply) {
-              DLOG_DEBUG("Got reply {}", reply->name()->str());
-              if (reply->name()->str() != client.ip_str) {
-                  LOG_THROW("Server did not see our IP {} != {}", reply->name()->str(), client.ip_str);
-              }
-              return seastar::make_ready_future<>();
-            })
-            .then([&client] { 
-              return client.client->stop(); 
-            });
+              .then([&client] {
+                DLOG_DEBUG("Connected to {}", client.client->server_addr);
+                smf::rpc_typed_envelope<smf_gen::demo::Request> req;
+                req.data->name =
+                  ipv4_addr_to_ip_str(client.client->server_addr);
+                return client.client->Get(req.serialize_data());
+              })
+              .then([&client](auto reply) {
+                DLOG_DEBUG("Got reply {}", reply->name()->str());
```

Comment:
store 2 vars, since you are already doing it. and print them in case of failure.